cmd/chain-auditor/main.go

```diff
@@ -159,26 +262,66 @@ func main() {
 	}()
 
 	var wg sync.WaitGroup
-	for i := 0; i < 50; i++ {
+	for i := 0; i < *parallelism; i++ {
 		wg.Add(1)
 		go func() {
 			for hostname := range hnChan {
-				mismatched := auditChainForHostname(hostname)
-				if mismatched == true {
-					resChan <- hostname
+				result := auditChainForHostname(hostname)
+				if result.mismatched == true {
+					if debugMode == true {
+						fmt.Printf("%+v\n", result)
+					}
+					resChan <- result
+					mismatchedCount++
 				}
+				hostnamesRemainCount--
+				if !result.reachable == true {
+					unreachableCount++
+					if result.probs.dnsErr == true {
+						dnsCount++
+					}
+					if result.probs.netErrTimeout == true {
+						timeoutCount++
+					}
+					if result.probs.netErrOther == true {
+						otherCount++
+					}
+				}
+				progressBar.TickAndUpdate(bar.Context{
+					bar.Ctx("mismatched", strconv.Itoa(mismatchedCount)),
+					bar.Ctx("remain", strconv.Itoa(hostnamesRemainCount)),
+					bar.Ctx("unreachable", strconv.Itoa(unreachableCount)),
+					bar.Ctx("dns", strconv.Itoa(dnsCount)),
+					bar.Ctx("timeout", strconv.Itoa(timeoutCount)),
+					bar.Ctx("other", strconv.Itoa(otherCount)),
+				})
 			}
 			wg.Done()
 		}()
 	}
 
 	go func() {
 		for result := range resChan {
-			fmt.Println(result)
+			_, err := auditFile.Write([]byte(fmt.Sprintf("%s\t%s\n", result.hostname, result.ip)))
```

Comment:
I would do this with `fmt.Fprintf`:

```suggestion
			err := fmt.Fprintf(auditFile, "%s\t%s\n", result.hostname, result.ip)
```

Reply:
🤦🏻 yeah, this is a lot cleaner too